starky/src/prover.rs

```diff
@@ -353,6 +358,10 @@ where
     let rate_bits = config.fri_config.rate_bits;
     let total_num_helper_cols: usize = num_ctl_columns.iter().sum();
 
+    if stark.quotient_degree_factor() == 0 {
+        return None;
+    }
+
```

Comment:
Nit: Can't this be at the very start at the function? (we don't need the degree, rate bits and total number of helper columns if we're not computing the polynomial)